packages/@lwc/template-compiler/src/parser/parser.ts

```diff
@@ -66,6 +69,7 @@ export default class ParserCtx {
      * Each scope corresponds to the original parse5.Element node.
      */
     private readonly scopes: ParentNode[][] = [];
```

Comment:
Some renaming is necessary here.

@jmsjtu I had some trouble determining what kind of scope these are supposed to be to distinguish them from `siblingScope`. Would it be `elementScope`? `nodeScope`?

Reply:
This specifically stores the `ForBlock`, `If`, `IfBlock`, `ElseBlock`, `Element`, `Component` and `Slot`. 

The type that encompasses all of these is `ParentNode`.

The intention of this array is to keep track of all the AST nodes that are generated from a single HTML Element.

I think `elementScope` would be appropriate in this case to help distinguish it from `siblingScope`.